lib/licensing.js

```diff
@@ -130,10 +130,56 @@ const listSubComponents = (baseDir) => {
   return subComponents
 }
 
+const listNtpBackgrounds = (metadataFile) => {
+  const metadata = fs.readFileSync(metadataFile).toString()
+  let images = []
+  // Hack to import this TypeScript file in Node:
+  eval(metadata.replace('export const images: NewTab.Image[]', 'images'))
+  return images
+}
+
+const getValidatedComponentField = (component, fieldName) => {
+    const fieldValue = component[fieldName]
+    if (!fieldValue) {
+       console.error('Missing' + fieldName + ' for background image ' + component['name'])
+       process.exit(1)
+    }
+    return fieldValue
+}
+
+const generateNtpBackgroundsLicense = (preamble, components) => {
+  let componentNotices = ''
+
+  for (let component of components) {
+    if (componentNotices.length != 0) {
+      componentNotices += '\n'
+    }
+
+    const fileName = getValidatedComponentField(component, 'source')
+    const authorName = getValidatedComponentField(component, 'author')
+    const authorLink = getValidatedComponentField(component, 'link')
+    const originalUrl = getValidatedComponentField(component, 'originalUrl')
+    const license = getValidatedComponentField(component, 'license')
+    if ((license != 'used with permission') &&
```

Comment:
cc @fmarier better if this could be case insensitive. had to workaround in https://github.com/brave/brave-core/pull/4864